### PR TITLE
Release 30

### PR DESCRIPTION
**Pull request by renanrodrigo**
This is the version 30 release of ubuntu-advantage-tools.

### CODE REVIEW

**Comment by renanrodrigo** [comment]
had to run the po script on top of the rebase

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,69 @@
...
+    - define the files to look for translatable messages
+
+  * d/po/pt_BR.po:
+    - add first Brazilian Portuguese translations
+
+  * d/po/templates.pot:
+    - remove old translation template for the Client
+
+  * d/po/ubuntu-pro.pot:
+    - add new translation template for the Client
```
Comment (orndorffgrant) [inline debian/changelog:21]: Mentioned out-of-band, but I think we can combine these into one entry
```suggestion
  * d/po:
    - Makefile to build localization files to debian/po/usr/share/locale/
    - update POTFILES.in to cover all translatable messages
    - remove old unused pot file
    - add new complete pot file for "ubuntu-pro" domain
    - add first Brazilian Portuguese translations
```

File: debian/changelog [hunk 2]
```diff
@@ -1,3 +1,69 @@
...
+
+  * d/ubuntu-advantage-tools.maintscript
+    - remove /etc/ubuntu-advantage/help_data.yaml
+
+  * d/ubuntu-pro-client-l10n.install
+    - add install file for the new binary package
+
+  * New upstream release 30 (LP: <TODO>)
+    - api:
+      + add new plan steps to the fix plan endpoints
```
Comment (orndorffgrant) [inline debian/changelog:37]:
```suggestion
      + add new backwards compatible plan steps to the v1 fix plan endpoints
```

File: debian/changelog [hunk 3]
```diff
@@ -1,3 +1,77 @@
...
+      + improve information returned from the fix plan endpoints
+      + new endpoint: u.pro.security.fix.cve.execute.v1
+      + new endpoint: u.pro.security.fix.usn.execute.v1
+    - apt: improve performance and consistency by refactoring the code to use
+      the apt_pkg module
+    - auto-attach: add newlines to the MOTD message
```
Comment (orndorffgrant) [inline debian/changelog:43]:
```suggestion
    - auto-attach: add newline to the MOTD message to separate it from other MOTD messages
```

File: debian/changelog [hunk 4]
```diff
@@ -1,3 +1,77 @@
...
+    - logging:
+      + add journald logging for the daemon and systemd timer
+      + remove daemon and timer log files
+      + standardize the logging calls through the codebase (GH: #2632)
+    - systemd: change ubuntu-advantage.service type from 'notify' to 'exec',
+      dropping the dependency on python3-systemd (LP: #2038417) (GH: #2692)
+    - tests:
+      + add scenarios where cloud-init is present but disabled (LP: #1938208)
+      + change 'permission' to 'priority' when checking apt priority in tests
+        (LP: #2719)
```
Comment (orndorffgrant) [inline debian/changelog:71]:
```suggestion
        (GH: #2719)
```

**Comment by renanrodrigo** [comment]
@orndorffgrant addressed

**Comment by orndorffgrant** [comment]
@renanrodrigo The update-pos was needed because of this slight change that must have happened over the course of one of the conflict resolutions
```diff
diff --git a/uaclient/messages/__init__.py b/uaclient/messages/__init__.py
index 0c0f5539..39a6f70e 100644
--- a/uaclient/messages/__init__.py
+++ b/uaclient/messages/__init__.py
@@ -2006,6 +2006,10 @@ E_PYCURL_ERROR = FormattedNamedMessage(
     "pycurl-error", t.gettext("PycURL Error: {e}")
 )
 
+E_PYCURL_CA_CERTIFICATES = NamedMessage(
+    "pycurl-ca-certificates-error", "Problem reading SSL CA certificates"
+)
+
 E_PROXY_AUTH_FAIL = NamedMessage(
     "proxy-auth-fail", t.gettext("Proxy authentication failed")
 )
@@ -2506,7 +2510,3 @@ E_INCORRECT_ENUM_VALUE_ERROR_MESSAGE = FormattedNamedMessage(
         "value: {values}"
     ),
 )
-
-E_PYCURL_CA_CERTIFICATES = NamedMessage(
-    "pycurl-ca-certificates-error", "Problem reading SSL CA certificates"
-)
```

I think if you add a commit on top that fixes that diff - then the update-pos won't be necessary, and then `release` and `main` can live happily in sync

**Review by orndorffgrant — APPROVED** [review]

File: debian/changelog [hunk 5]
```diff
@@ -1,3 +1,70 @@
...
+    - contract: send information about variants to the contracts server
+    - enable: update only service specific apt sources when enabling a service
+      (GH: #1311) (GH: #1482)
+    - esm: create static files to pin packages from esm-infra and esm-apps with
+      higher priority (GH: #2580)
+    - disable:
+      + add the --purge flag to the disable command, so users can remove all
+        service related packages when disabling a service
+      + show extra warnings when kernels are involved in the purge operation
+    - files: Reduce race window when creating new files (LP: #2024204)
```
[the remark this answers is not in the document]
Comment (renanrodrigo) [reply debian/changelog:46]: Yes it is. Explainable out of band.

File: debian/changelog [hunk 6]
```diff
@@ -1,3 +1,70 @@
...
+    - fips: add support to Jammy to prepare for when it is available
+    - fips-preview:
+      + add fips-preview as a new entitlement
+    - github:
```
[the remark this answers is not in the document]
Comment (renanrodrigo) [reply debian/changelog:50]: we don't, but this points to a bug we fixed, and we agreed on listing all fixed bugs

File: debian/changelog [hunk 7]
```diff
@@ -1,3 +1,70 @@
...
+    - internationalization:
+      + add general internationalization support and templates
+      + add initial sentence set for Brazilian Portuguese
+    - logging:
+      + add journald logging for the daemon and systemd timer
+      + remove daemon and timer log files
+      + standardize the logging calls through the codebase (GH: #2632)
+    - systemd: change ubuntu-advantage.service type from 'notify' to 'exec',
+      dropping the dependency on python3-systemd (LP: #2038417) (GH: #2692)
+    - tests:
```
[the remark this answers is not in the document]
Comment (renanrodrigo) [reply debian/changelog:61]: same bug consideration. We are closing issues, and thus mentioning tests to list them

[the remark this answers is not in the document]
Comment (lucasmoura) [reply debian/changelog:60]: Fair enough, but I think the github one can be deleted.
Also, I think we can remove the github section too as it is not tied to package at all

[the remark this answers is not in the document]
Comment (renanrodrigo) [reply debian/changelog:60]: We will rediscuss the way we list the bugs in the changelogs.
This one may stay as it was a user request, and can be listed there without concerns, as agreed.

**Pull request merged**